### PR TITLE
DOC: Added document for DCO and commit message

### PR DESCRIPTION
**Pull request by Madhu-1**
# Describe what this PR does #

Added document on the standard user needs to follow when writing the commit message and to include sign-off in the commit message.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

## Related issues ##

Updates #773

### CODE REVIEW

File: docs/development-guide.md [hunk 1]
```diff
@@ -76,6 +76,67 @@ for code contributions.
...
+git commit --amend -s
+```
+
+We use a [DCO bot](https://github.com/apps/dco) to enforce the DCO on each pull
+request and branch commits.
+
+#### Commit Messages
+
+We follow a rough convention for commit messages that is designed to answer two
+questions: what changed and why. The subject line should feature the what and
```
Comment (humblec) [inline docs/development-guide.md:116]: what chan.... and why ?

File: docs/development-guide.md [hunk 2]
```diff
@@ -76,6 +76,67 @@ for code contributions.
 
 Please read the [coding guidelines](coding.md) document before submitting a PR.
 
+#### Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
```
Comment (humblec) [inline docs/development-guide.md:82]: DOC

Comment (Madhu-1) [reply docs/development-guide.md:82]: it should be DCO. updated now

**Comment by nixpanic** [comment]
Where does the text come from? A pointer to the source in the commit message would be good to have.

**Comment by Madhu-1** [comment]
> Where does the text come from? A pointer to the source in the commit message would be good to have.

done, added the source

**Comment by Madhu-1** [comment]
@nixpanic I updated some minor data in development doc, can you please review it once again
@humblec PTAL

**Pull request merged**